pom.xml

```diff
@@ -125,8 +127,15 @@
     <maven.javadoc.skip>true</maven.javadoc.skip>
     <maven.build.timestamp.format>yyyy-MM-dd'T'HH:mm</maven.build.timestamp.format>
     <buildDate>${maven.build.timestamp}</buildDate>
-    <compileSource>1.8</compileSource>
-    <java.min.version>${compileSource}</java.min.version>
+    <!-- Project level compilation properties -->
+    <compileSource>17</compileSource>
```

Comment:
We'd better use `--release 8` for all modules expect hbase-unsafe. For hbase-unsafe, we should use JDK8 to compile it, and set compileSource and compileTarget to 8.